src/javascript/SelectorTypes/Picker/Picker2.redux.js

```diff
@@ -138,7 +138,14 @@ export const registerPickerReducer = registry => {
         }),
         [cePickerSetSelection]: (state, action) => ({
             ...state,
-            selection: action.payload
+            selection: action.payload,
+            searchPath: '',
+            searchTerms: '',
+            mode: state.mode === Constants.mode.SEARCH ? state.preSearchModeMemo : state.mode,
+            pagination: {
+                ...state.pagination,
+                currentPage: 0
+            }
```

Comment:
i don't understand why resetting search here in cePickerSetSelection - this one is called when opening a picker, to set the previous selection (and so not called when there's no previous selection)
cePickerSetSearchTerm is already called in toggleOpen  ?

Reply:
cePickerSetSelection is called all the time it seems (with []), toggleOpen I did not noticed as in redux all I saw was the call to cePickerKey. The search is not emptied upon opening (maybe an issue with batch?)

Reply:
cePickerKey is called when opening, the batch is called when closing.
I find strange to reset search when calling cePickerSetSelection - this one could be called anytime else than when opening the picker

Reply:
`toggleOpen` is not called when closing editor but `onItemSelection` instead when there is a selection. `onItemSelection` then calls `setDialogOpen` directly to close the dialog. Perhaps `onItemSelection` should call `toggleOpen` instead to trigger the batch actions.

Either that or call the batch actions to clear selection/search term in the else clause instead (when editor is being opened)

Reply:
right, it would be better to call toggleOpen() . We can also reset the searchTerms when both opening and closing , not only closing ?

Reply:
will reset on opening too as it makes